sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsAsyncTests.java

```diff
@@ -89,4 +92,153 @@ public void componentLifecycleTest(HttpClient httpClient, DigitalTwinsServiceVer
+
+            // Update the component again, but with the out of date etag
+            StepVerifier.create(
+                asyncClient.updateComponentWithResponse(
+                    roomWithWifiTwinId,
+                    wifiComponentName,
+                    TestAssetsHelper.getWifiComponentSecondUpdatePayload(),
+                    new UpdateComponentOptions().setIfMatch(etagBeforeUpdate.get())))
+                .verifyErrorSatisfies(ex -> assertRestException(ex, HttpURLConnection.HTTP_PRECON_FAILED));
+        }
+        finally {
+            try
+            {
+                if (roomWithWifiTwinId != null)
+                {
+                    asyncClient.deleteDigitalTwin(roomWithWifiTwinId).block();
+                }
+                if (roomWithWifiModelId != null)
+                {
+                    asyncClient.deleteModel(roomWithWifiModelId).block();
+                }
+                if (wifiModelId != null)
+                {
+                    asyncClient.deleteModel(wifiModelId).block();
+                }
+            }
+            catch (Exception ex)
+            {
+                fail("Test cleanup failed", ex);
+            }
+        }
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.digitaltwins.core.TestHelper#getTestParameters")
+    @Override
+    public void patchComponentSucceedsIfETagMatches(HttpClient httpClient, DigitalTwinsServiceVersion serviceVersion) throws JsonProcessingException {
+        DigitalTwinsAsyncClient asyncClient = getAsyncClient(httpClient, serviceVersion);
+
+        String wifiComponentName = "wifiAccessPoint";
+
+        String roomWithWifiTwinId = UniqueIdHelper.getUniqueDigitalTwinId(TestAssetDefaults.ROOM_WITH_WIFI_TWIN_ID_PREFIX, asyncClient, randomIntegerStringGenerator);
+        String roomWithWifiModelId = UniqueIdHelper.getUniqueModelId(TestAssetDefaults.ROOM_WITH_WIFI_MODEL_ID_PREFIX, asyncClient, randomIntegerStringGenerator);
+        String wifiModelId = UniqueIdHelper.getUniqueModelId(TestAssetDefaults.WIFI_MODEL_ID_PREFIX, asyncClient, randomIntegerStringGenerator);
+
+        String modelWifi = TestAssetsHelper.getWifiModelPayload(wifiModelId);
+        String modelRoomWithWifi = TestAssetsHelper.getRoomWithWifiModelPayload(roomWithWifiModelId, wifiModelId, wifiComponentName);
+        String roomWithWifiTwin = TestAssetsHelper.getRoomWithWifiTwinPayload(roomWithWifiModelId, wifiComponentName);
+        List<String> modelsList = new ArrayList<>(Arrays.asList(modelWifi, modelRoomWithWifi));
+
+        try {
+            // Create models and components to test the lifecycle.
+            StepVerifier
+                .create(asyncClient.createModels(modelsList))
+                .assertNext(createResponseList -> logger.info("Created models successfully"))
+                .verifyComplete();
+
+            StepVerifier.create(asyncClient.createOrReplaceDigitalTwinWithResponse(
```

Comment:
formatting differences between this and the previous call on line 189 in terms of line breaks and tabbing.